integration_tests/validator_out.go

```diff
@@ -0,0 +1,169 @@
+			s.Require().NoError(err)
+
+			bankQueryClient := banktypes.NewQueryClient(clientCtx)
+			res, err := bankQueryClient.AllBalances(context.Background(),
+				&banktypes.QueryAllBalancesRequest{
+					Address: s.chain.validators[1].keyInfo.GetAddress().String(),
+				})
+			if err != nil {
+				return false
+			}
+
+			gbQueryClient := types.NewQueryClient(clientCtx)
+			denomRes, err := gbQueryClient.ERC20ToDenom(context.Background(),
+				&types.ERC20ToDenomRequest{
+					Erc20: testERC20contract.String(),
+				})
+			if err != nil {
+				s.T().Logf("error querying ERC20 denom %s, %e", testERC20contract.String(), err)
+				return false
+			}
+			s.Require().False(denomRes.CosmosOriginated, "ERC20-originated denom marked as cosmos originated")
+			gravityDenom = denomRes.Denom
+
+			for _, coin := range res.Balances {
+				if coin.Denom == gravityDenom && coin.Amount.Equal(sdk.NewInt(200)) {
+					return true
+				}
+			}
+
+			s.T().Logf("balance not found, received %v", res.Balances)
+
+			return false
+		}, 105*time.Second, 10*time.Second, "balance never found on cosmos")
+
+		s.T().Logf("sending to ethereum")
+		sendToEthereumMsg := types.NewMsgSendToEthereum(
+			s.chain.validators[1].keyInfo.GetAddress(),
+			s.chain.validators[1].ethereumKey.address,
+			sdk.Coin{Denom: gravityDenom, Amount: sdk.NewInt(100)},
+			sdk.Coin{Denom: gravityDenom, Amount: sdk.NewInt(1)},
+		)
+
+		// Send NewMsgSendToEthereum Message
+		s.Require().Eventuallyf(func() bool {
+			val := s.chain.validators[1]
+			keyring, err := val.keyring()
+			s.Require().NoError(err)
+			clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &keyring, "val", val.keyInfo.GetAddress())
+			s.Require().NoError(err)
+
+			response, err := s.chain.sendMsgs(*clientCtx, sendToEthereumMsg)
+			if err != nil {
+				s.T().Logf("error: %s", err)
+				return false
+			}
+			if response.Code != 0 {
+				if response.Code != 32 {
+					s.T().Log(response)
+				}
+				return false
+			}
+			return true
+		}, 20*time.Minute, 10*time.Second, "unable to send to ethereum")
+
+		// Create Transaction batch
+		s.Require().Eventuallyf(func() bool {
+			batchTx := types.NewMsgRequestBatchTx(gravityDenom, s.chain.validators[2].keyInfo.GetAddress())
+
+			keyRing, err := s.chain.validators[2].keyring()
+			s.Require().NoError(err)
+
+			clientCtx, err := s.chain.clientContext("tcp://localhost:26657", &keyRing, "val", s.chain.validators[2].keyInfo.GetAddress())
+			s.Require().NoError(err)
+
+			response, err := s.chain.sendMsgs(*clientCtx, batchTx)
+			s.T().Logf("batch response: %s", response)
+			if err != nil {
+				s.T().Logf("error: %s", err)
+				return false
+			}
+
+			if response.Code != 0 {
+				if response.Code != 32 {
+					s.T().Log(response)
+				}
+				return false
+			}
+
+			s.Require().NoError(err, "error querying delegator bonded validators")
+
+			return true
+		}, 30*time.Minute, 1*time.Second, "can't create TX batch successfully")
```

Comment:
Same here, this test might block others unless 30 minutes is a reasonable wait

Reply:
done!